api/provider/aws/common.go

```diff
@@ -278,3 +279,27 @@ func readService(ecsapi ecsiface.ECSAPI, clusterName, serviceID string) (*ecs.Se
 
 	return output.Services[0], nil
 }
+
+// Check for eventually consistency
+func waitTillSGDeleted(ec2api ec2iface.EC2API, securityGroupName string) (shouldRetry bool, err error) {
+	filter := &ec2.Filter{}
+	filter.SetName("group-name")
+	filter.SetValues([]*string{aws.String(securityGroupName)})
+
+	input := &ec2.DescribeSecurityGroupsInput{}
+	input.SetFilters([]*ec2.Filter{filter})
+
+	output, err := ec2api.DescribeSecurityGroups(input)
+	if err != nil && !strings.Contains(err.Error(), "does not exist") {
```

Comment:
I'm worried that in the "pass-through" case  (err == some error with "does not exist" message), `output` will be `nil`. If that happens, line 297 `range output.SecurityGroups` will cause a panic. I think the safer approach here is:
```go
output, err := ec2api.DescribeSecurityGroups(input)
if err != nil {
    if strings.Contains(err.Error(), "does not exist") {
        return false, nil
    }

    return false, err
}
```